### PR TITLE
RemoteControl application

### PR DESCRIPTION
**Pull request by gabryelreyes**

### CODE REVIEW

File: lib/RemoteControl/src/App.h [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+
+/** The convoy follower application. */
+class App
```
Comment (BlueAndi) [inline lib/RemoteControl/src/App.h:59]: Wrong description

File: lib/RemoteControl/src/App.h [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+{
+public:
+    /**
+     * Construct the convoy follower application.
```
Comment (BlueAndi) [inline lib/RemoteControl/src/App.h:63]: Wrong description

File: lib/RemoteControl/src/App.h [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+     */
+    App() :
+        m_smpServer(Board::getInstance().getDevice().getStream()),
+        m_serialMuxProtChannelIdRemoteCtrl(0U),
+        m_serialMuxProtChannelIdMotorSpeeds(0U)
+    {
+    }
+
+    /**
+     * Destroy the convoy follower application.
```
Comment (BlueAndi) [inline lib/RemoteControl/src/App.h:73]: Wrong description

File: lib/RemoteControl/src/App.cpp [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+ * @param[in] payload       Command id
+ * @param[in] payloadSize   Size of command id
+ */
+void App_cmdRspChannelCallback(const uint8_t* payload, const uint8_t payloadSize)
+{
+    LOG_DEBUG("CMD_RSP: 0x%02X", payload[0]);
+}
+
+/**
+ * Receives line sensor data over SerialMuxProt channel.
```
Comment (BlueAndi) [inline lib/RemoteControl/src/App.cpp:303]: Wrong description.

File: lib/RemoteControl/src/App.cpp [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+ * @param[in]   payload         Line sensor data
+ * @param[in]   payloadSize     Size of 5 line sensor data
+ */
+void App_lineSensorChannelCallback(const uint8_t* payload, const uint8_t payloadSize)
+{
+    ; /* Do nothing */
```
Comment (BlueAndi) [inline lib/RemoteControl/src/App.cpp:309]: Please set function arguments as not used via cast.

File: lib/RemoteControl/src/App.cpp [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+
+/**
+ * Receives remote control commands over SerialMuxProt channel.
+ *
+ * @param[in] payload       Command id
+ * @param[in] payloadSize   Size of command id
+ */
+void App_cmdRspChannelCallback(const uint8_t* payload, const uint8_t payloadSize)
+{
+    LOG_DEBUG("CMD_RSP: 0x%02X", payload[0]);
```
Comment (BlueAndi) [inline lib/RemoteControl/src/App.cpp:299]: Payload size not checked.

File: lib/RemoteControl/src/App.cpp [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+    if (error != DeserializationError::Ok)
+    {
+        LOG_ERROR("JSON Deserialization Error %d.", error);
+    }
+    else
+    {
+        uint8_t payloadSize = sizeof(uint8_t);
+        uint8_t buffer[payloadSize];
+
+        buffer[0U] = jsonPayload["CMD_ID"].as<uint8_t>();
```
Comment (BlueAndi) [inline lib/RemoteControl/src/App.cpp:240]: Is CMD_ID really inside the jsonPayload? What happens if not?

File: lib/RemoteControl/src/App.cpp [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+    {
+        LOG_ERROR("JSON Deserialization Error %d.", error);
+    }
+    else
+    {
+        uint8_t payloadSize = (2U * sizeof(int16_t));
+        uint8_t buffer[payloadSize];
+
+        Util::int16ToByteArray(&buffer[0U * sizeof(int16_t)], sizeof(int16_t), jsonPayload["LEFT"].as<int16_t>());
+        Util::int16ToByteArray(&buffer[1U * sizeof(int16_t)], sizeof(int16_t), jsonPayload["RIGHT"].as<int16_t>());
```
Comment (BlueAndi) [inline lib/RemoteControl/src/App.cpp:270]: Is LEFT/RIGHT really inside the jsonPayload? What happens if not?

File: lib/RemoteControl/library.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    "authors": [{
+        "name": "Gabryel Reyes",
+        "email": "gabryelrdiaz@gmail.com",
+        "url": "https://github.com/gabryelreyes",
+        "maintainer": true
+    }],
+    "license": "MIT",
+    "dependencies": [{
+        "name": "SerialMuxProt",
+        "version" : "https://github.com/gabryelreyes/SerialMuxProt.git"
```
Comment (BlueAndi) [inline lib/RemoteControl/library.json:14]: Using a fixed version would be better.

**Review by BlueAndi — CHANGES_REQUESTED** [review]

File: lib/RemoteControl/src/App.cpp (hunk 4 above)
Comment (gabryelreyes) [reply lib/RemoteControl/src/App.cpp:303]: Really? It is the callback for SMP. Why is it wrong?

**Review by BlueAndi — APPROVED** [review]

**Pull request merged**